native/acter/src/api/reactions.rs

```diff
@@ -0,0 +1,218 @@
+}
+
+impl Deref for ReactionManager {
+    type Target = models::ReactionManager;
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl ReactionManager {
+    pub(crate) fn new(
+        client: Client,
+        room: Room,
+        inner: models::ReactionManager,
+    ) -> ReactionManager {
+        ReactionManager {
+            client,
+            room,
+            inner,
+        }
+    }
+
+    pub async fn send_reaction(&self, event_id: String, key: String) -> Result<OwnedEventId> {
+        let room = self.room.clone();
+        let client = room.client();
+        let my_id = client.user_id().context("User not found")?.to_owned();
+        let event_id = EventId::parse(&event_id)?;
+
+        RUNTIME
+            .spawn(async move {
+                let member = room
+                    .get_member(&my_id)
+                    .await?
+                    .context("Couldn't find me among room members")?;
+                if !member.can_send_message(MessageLikeEventType::RoomMessage) {
+                    bail!("No permission to send message in this room");
+                }
+
+                trace!("before sending reaction");
+                let content = ReactionEventContent::new(Annotation::new(event_id, key));
+                let response = room.send(content).await?;
+
+                trace!("after sending reaction");
+                Ok(response.event_id)
+            })
+            .await?
+    }
+
+    pub async fn redact_reaction(
+        &self,
+        event_id: String,
+        reason: Option<String>,
+        txn_id: Option<String>,
+    ) -> Result<OwnedEventId> {
+        let room = self.room.clone();
+        let my_id = room
+            .client()
+            .user_id()
+            .context("User not found")?
+            .to_owned();
+        let event_id = EventId::parse(event_id)?;
+        let txn_id = match txn_id {
+            Some(x) => Some(OwnedTransactionId::try_from(x)?),
+            None => None,
+        };
+
+        RUNTIME
+            .spawn(async move {
+                let member = room
+                    .get_member(&my_id)
+                    .await?
+                    .context("Couldn't find me among room members")?;
+                if !member.can_send_message(MessageLikeEventType::RoomMessage) {
+                    bail!("No permission to send message in this room");
+                }
+                trace!("before redacting reaction");
+                let response = room.redact(&event_id, reason.as_deref(), txn_id).await?;
+                trace!("after redacting reaction");
+                Ok(response.event_id)
+            })
+            .await?
+    }
+
+    pub fn stats(&self) -> models::ReactionStats {
+        self.inner.stats().clone()
+    }
+
+    pub async fn my_status(&self) -> Result<bool> {
+        let manager = self.inner.clone();
+        let my_id = self.client.user_id().context("User not found")?;
```

Comment:
considering how often the user_id is needed, we might want to consider loading that once at creation time and just store it in the struct, too...